tools/codegen/model.py

```diff
@@ -199,6 +199,54 @@ def __post_init__(self) -> None:
 
 SchemaKind = Enum('SchemaKind', ('functional', 'inplace', 'out'))
 
+# Represents a bundle of native functions that are semantically related.
+@dataclass(frozen=True)
+class NativeFunctionGroup:
```

Comment:
A third concept we also name "group"?

Reply:
Second, really; after I got rid of `grouped_arguments` for packs in faithful C++ signature we are now just "group arguments" and "group native functions." IDK if you have a better name; in this case it seems less bad because the entity being grouped is different between the two cases.